applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -399,65 +399,125 @@ namespace Kratos {
         double& added_search_distance = r_process_info[SEARCH_RADIUS_INCREMENT];
+        const bool local_coordination_option = r_process_info[LOCAL_COORDINATION_NUMBER_OPTION];
+        const bool global_coordination_option = r_process_info[GLOBAL_COORDINATION_NUMBER_OPTION];
         added_search_distance = 0.0;
         double amplification = 1.0;
 
         if(r_model_part.GetCommunicator().MyPID() == 0) {
-            KRATOS_WARNING("DEM") << "Setting up Coordination Number by increasing or decreasing the search radius... " << std::endl;
-        }
+            KRATOS_INFO("DEM") << "Setting up Coordination Number (input = "<<desired_coordination_number<<") by increasing or decreasing the search radius. ";
+            if(local_coordination_option) KRATOS_INFO("") << "Local extension activated. ";
+            if(global_coordination_option) KRATOS_INFO("") << "Global extension activated. ";
+            KRATOS_INFO("DEM") << std::endl;
+        }
+
+        if (desired_coordination_number <= 0.0) {
+            KRATOS_THROW_ERROR(std::runtime_error, "The specified Coordination Number is less or equal to zero, N.C. = ", desired_coordination_number)
+        }
+
+        //STAGE 1, Local Coordination Number
+        double tolerance = 0.5;
+        double max_factor_between_iterations = 1.02;
+        double relative_error = 1000;
+        if(local_coordination_option) {
+            while (std::abs(relative_error) > tolerance) {
+                if (iteration >= maxiteration) break;
+                iteration++;
+                if (current_coordination_number == 0.0) {
+                    KRATOS_WARNING("DEM") << "Coordination Number method not supported in this case" << "\n" << std::endl;
+                    KRATOS_THROW_ERROR(std::runtime_error, "The specified tangency method is not supported for this problem, please use absolute value instead", " ")
+                    break;
+                }
 
-        if (in_coordination_number <= 0.0) {
-            KRATOS_THROW_ERROR(std::runtime_error, "The specified Coordination Number is less or equal to zero, N.C. = ", in_coordination_number)
-        }
-        while (fabs(out_coordination_number / in_coordination_number - 1.0) > 1e-3) {
-            if (iteration >= maxiteration) break;
-            iteration++;
-            if (out_coordination_number == 0.0) {
-                KRATOS_WARNING("DEM") << "Coordination Number method not supported in this case" << "\n" << std::endl;
-                KRATOS_THROW_ERROR(std::runtime_error, "The specified tangency method is not supported for this problem, please use absolute value instead", " ")
-                break;
-            }
+                std::vector<double> total_error;
+                mNumberOfThreads = OpenMPUtils::GetNumThreads();
+                total_error.resize(mNumberOfThreads);
+
+                //#pragma omp parallel for
+                for (size_t i = 0; i < mListOfSphericContinuumParticles.size(); i++) {
```

Comment:
```suggestion
                for (int i = 0; i < static_cast<int>(mListOfSphericContinuumParticles.size()); ++i) {
```

otherwise will not work in Win (if you enable omp)